api/v3/Activity.php

```diff
@@ -304,11 +304,23 @@ function civicrm_api3_activity_get($params) {
         "Cannot access activities. Required permission: 'view all activities''"
       );
     }
-
-    if (!CRM_Activity_BAO_Activity::checkPermission($params['id'], CRM_Core_Action::VIEW)) {
-      throw new \Civi\API\Exception\UnauthorizedException(
-        'You do not have permission to view this activity'
-      );
+    $ids = array();
+    if (is_array($params['id'])) {
+      foreach ($params['id'] as $operator => $values) {
+        if (in_array($operator, CRM_Core_DAO::acceptedSQLOperators())) {
```

Comment:
This could result in permission escalation if the operator is not "IN". E.g. passing `'id' => array('NOT IN' => 123)` would let you see all activities if you have access to view 123.